R/mainTests.r

```diff
@@ -1498,6 +1953,39 @@ filter_group_duplicate <- function(meta) {
   return(output)
 }
 
+# whitespace_filters -------------------------------------
+# check no filter labels have leading or trailing whitespace.
+
+whitespace_filters <- function(data, meta) {
+  filters <- meta %>%
+    filter(col_type == "Filter") %>%
+    pull(col_name)
+
+  test <- data %>%
+    select(all_of(filters), any_of(as.character(geography_matrix[, 2]))) %>%
+    gather(., "filter", "filter_label") %>%
+    distinct()
+
+  test_no_whitespace <- test %>% mutate(filter_label = str_trim(filter_label))
+
+  white_spaces <- setdiff(test, test_no_whitespace) %>% pull(filter_label)
+
+  if (length(white_spaces) == 0) {
+    output <- list(
+      "message" = "No filter labels contain leading or trailing whitespace.",
+      "result" = "PASS"
+    )
+  } else {
```

Comment:
Could add a singular version for where `length(white_spaces == 1)`?

Reply:
Yes added now :)